packages/server/src/asyncQuestion/asyncQuestion.controller.ts

```diff
@@ -150,8 +158,98 @@ export class asyncQuestionController {
+  @Patch('student/:questionId')
+  /**
+   * Updates a student's async question.
+   *
+   * @param {number} questionId - The ID of the question to update.
+   * @param {UpdateAsyncQuestions} body - The updated question data.
+   * @param {UserModel} user - The user making the request.
+   * @return {Promise<AsyncQuestionParams>} The updated question.
+   * @throws {NotFoundException} If the question is not found.
+   * @throws {HttpException} If the user is not the creator of the question.
+   * @throws {HttpException} If the user tries to update the question's status to TADeleted or HumanAnswered.
+   */
+  async updateStudentQuestion(
+    @Param('questionId') questionId: number,
+    @Body() body: UpdateAsyncQuestions,
+    @User() user: UserModel,
+  ): Promise<AsyncQuestionParams> {
+    const question = await AsyncQuestionModel.findOne({
+      where: { id: questionId },
+      relations: ['course', 'creator', 'votes'],
+    });
+
+    if (!question) {
+      throw new NotFoundException();
+    }
+
+    if (question.creatorId !== user.id) {
+      throw new HttpException(
+        'You can only update your own questions',
+        HttpStatus.UNAUTHORIZED,
+      );
+    }
+    if (body.status === asyncQuestionStatus.AIAnsweredNeedsAttention) {
+      const courseId = question.course.id;
+
+      // Step 1: Get all users in the course
+      const usersInCourse = await UserCourseModel.createQueryBuilder(
+        'userCourse',
+      )
+        .select('userCourse.userId')
+        .where('userCourse.courseId = :courseId', { courseId })
+        .getMany();
+
+      const userIds = usersInCourse.map((uc) => uc.userId);
+
+      // Step 2: Get subscriptions for these users
+      const subscriptions = await UserSubscriptionModel.createQueryBuilder(
+        'subscription',
+      )
+        .innerJoinAndSelect('subscription.user', 'user')
+        .innerJoinAndSelect('subscription.service', 'service')
+        .where('subscription.userId IN (:...userIds)', { userIds })
+        .andWhere('service.serviceType = :serviceType', {
+          serviceType: MailServiceType.ASYNC_QUESTION_FLAGGED,
+        })
+        .andWhere('subscription.isSubscribed = true')
+        .getMany();
+
+      // Send emails in parallel
+      await Promise.all(
+        subscriptions.map((sub) =>
+          this.mailService.sendEmail({
+            receiver: sub.user.email,
+            type: MailServiceType.ASYNC_QUESTION_FLAGGED,
+            subject: 'UBC Helpme - Subscription Update',
+          }),
+        ),
+      );
+    }
+
+    // Update allowed fields
+    Object.keys(body).forEach((key) => {
+      if (body[key] !== undefined && body[key] !== null) {
+        question[key] = body[key];
+      }
+    });
+
+    const updatedQuestion = await question.save();
+
+    // Update Redis queue
+    await this.redisQueueService.updateAsyncQuestion(
+      `c:${question.course.id}:aq`,
+      updatedQuestion,
+    );
+
+    delete question.votes;
```

Comment:
```suggestion
    delete question.taHelped;
    delete question.votes;
```

The other endpoint has this, figured you might want to add it to this endpoint too